src/Microsoft.PowerShell.ConsoleHost/host/msh/StartTranscriptCmdlet.cs

```diff
@@ -228,7 +227,19 @@ protected override void BeginProcessing()
                         System.IO.File.Delete(effectiveFilePath);
                     }
                 }
-
+                else 
+                {
+                    // check if the path resolves to an existing or non-existing directory. 
+                    string baseDirectory = System.IO.Path.GetDirectoryName(effectiveFilePath);
+                    if(System.IO.Directory.Exists(effectiveFilePath) || (string.Compare(baseDirectory, System.IO.Path.GetFullPath(effectiveFilePath).TrimEnd('\\')) == 0))
```

Comment:
Very good point.. I switched to Path.DirectorySeparatorChar rather than hardcoding '\\'.
Path.GetFileName(Path) returns the dir name in the following scenario :
Path.GetFileName("C:\ExistingFolder") returns "ExistingFolder"
which is incorrect. We want to make sure the path does not resolve to a directory.